include/fs.h

```diff
@@ -8,23 +8,40 @@
 #define _FS_H_
 
 #include <sys/types.h>
-#include <fs/fs_interface.h>
+#include <misc/dlist.h>
+
+#ifdef CONFIG_FAT_FILESYSTEM_ELM
```

Comment:
can we avoid knowledge about the implementation of the various filesystems in the public header and do this on the filesystem implementation? This does not scale really well if we have to add this type of checks for every file system we add.

Reply:
@nashif  I have thought through this while implementing it. Right now we don't do much in our fs interface modules (subsys/fs/fat_fs.c) other than redirecting the calls to FS libs and the way we are handling things inside these modules are not uniform across file systems. I will relook into this once I am done with enabling multiple partition support for which I am working on disk susbsys.

For now, I will move the file system specific definitions info  fs_interface.h like the previous way and submit the PR. At least the public fs.h file will not have FS specific definitions.